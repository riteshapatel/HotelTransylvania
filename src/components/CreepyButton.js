/**
 * @class CreepyButton component 
 * @author ritesh.patel
 */
import React, {Component} from 'react';
import {Text, Button, TouchableOpacity} from 'react-native';
import styles from '../styles/Global.style.js';

export default class CreepyButton extends Component {
    render () {
        const data = this.props.data || {}
        return (
            <TouchableOpacity onPress={() => this.props.navigate(this.props.navigateTo, data)} style={styles.creepy_button}>
                <Text style={styles.creepy_text}>{this.props.title}</Text>
            </TouchableOpacity>                    
        )
    }
}