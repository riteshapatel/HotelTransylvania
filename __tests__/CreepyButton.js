/**
 * Unit test for CreepyButton component (Snapshot Comparison)
 * @author ritesh.patel
 */
import React from 'react';
import CreepyButton from '../src/components/CreepyButton';
import renderer from 'react-test-renderer';

test ('snapshot CreepyButton component', () => {
    const navigate = { navigate: jest.fn() }
    const tree = renderer.create(<CreepyButton title='title' navigateTo='navigateTo' navigate={navigate}/>).toJSON();
    expect(tree).toMatchSnapshot();
});