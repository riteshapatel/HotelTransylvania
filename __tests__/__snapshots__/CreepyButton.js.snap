// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot CreepyButton component 1`] = `
<View
  accessible={true}
  isTVSelectable={true}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "backgroundColor": "#640544",
      "borderColor": "#404040",
      "borderRadius": 4,
      "borderWidth": 1,
      "marginBottom": 5,
      "opacity": 1,
      "padding": 5,
      "textAlign": "center",
    }
  }
>
  <Text
    style={
      Object {
        "color": "white",
        "fontSize": 14,
        "fontWeight": "bold",
        "textAlign": "center",
      }
    }
  >
    title
  </Text>
</View>
`;
